server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -191,14 +190,7 @@ public Response doPost(
         queryId = UUID.randomUUID().toString();
         query = query.withId(queryId);
       }
-      if (query.getContextValue(QueryContextKeys.TIMEOUT) == null) {
```

Comment:
Instead of changing the default behavior, how about keeping the default behavior the same, but fixing the docs and making it more configurable? So I'm suggesting:

- Add a `druid.server.http.defaultQueryTimeout` property and document that in configuration/broker.md and configuration/historical.md. Use that to set the default timeout here, and default that property to PT5M.
- Update query-context.md to reflect that the default isn't 0, it's the value of `druid.server.http.defaultQueryTimeout`.